CDS/src/org/icpc/tools/cds/CDSAuth.java

```diff
@@ -0,0 +1,146 @@
+package org.icpc.tools.cds;
+
+import java.util.Base64;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.security.enterprise.AuthenticationException;
+import javax.security.enterprise.AuthenticationStatus;
+import javax.security.enterprise.authentication.mechanism.http.HttpAuthenticationMechanism;
+import javax.security.enterprise.authentication.mechanism.http.HttpMessageContext;
+import javax.security.enterprise.credential.UsernamePasswordCredential;
+import javax.security.enterprise.identitystore.CredentialValidationResult;
+import javax.security.enterprise.identitystore.CredentialValidationResult.Status;
+import javax.security.enterprise.identitystore.IdentityStoreHandler;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.icpc.tools.contest.Trace;
+import org.icpc.tools.contest.model.IAccount;
+
+@ApplicationScoped
+public class CDSAuth implements HttpAuthenticationMechanism {
+	private static final String AUTHORIZATION_HEADER = "Authorization";
+	private static final String BASIC_PREFIX = "Basic ";
+	private static final String RESULT = "result";
+	private static final String ACCOUNT = "account";
+
```

Comment:
So perhaps we should move those constants here?

Reply:
As above, went with IAccount since these are account types.